client/app/bundles/course/assessment/question/programming/ProgrammingForm.tsx

```diff
@@ -57,7 +60,7 @@ const ProgrammingForm = (props: ProgrammingFormProps): JSX.Element => {
 
       const toastSuccessAndRedirect = (): void => {
         toast.success(t(translations.questionSavedRedirecting));
-        window.location.href = response.redirectAssessmentUrl;
+        navigate(response.redirectAssessmentUrl);
```

Comment:
Any reason why we switch to `navigate` here but still use `window.location.href` in other parts of the code?

Reply:
Since we are already using React Router, generally, the preferred way of navigating should now be using `navigate`. `window.location.href` will trigger a refresh and should no longer be used for simple navigations.

Though, there are some instances where we need `window.location.href`. For example, downloading a file, triggering a simple GET, or forcing a full session reset. If you're referring to the other `window.location.href` use in this file, you'd quickly realise that it's not to navigate, but to jump to a section in the page by appending a [fragment identifier](https://www.w3.org/DesignIssues/Fragment.html).